crmd/lrm.c

```diff
@@ -2077,10 +2078,15 @@ do_lrm_rsc_op(lrm_state_t * lrm_state, lrmd_rsc_info_t * rsc, const char *operat
         }
     }
 
+    versioned_params = find_xml_node(msg, XML_TAG_VER_ATTRS, FALSE);
+    if (versioned_params) {
+        versioned_params = copy_xml(versioned_params);
+    }
+
     call_id = lrm_state_exec(lrm_state,
                              rsc->id,
                              op->op_type,
-                             op->user_data, op->interval, op->timeout, op->start_delay, params);
+                             op->user_data, op->interval, op->timeout, op->start_delay, params, versioned_params);
```

Comment:
lrm_state_exec() and lrmd_t's exec() function are part of the public API, so we want to avoid changing their arguments, which would break backward API/ABI compatibility.

One workaround we often use in such a case is to define a new function, for example lrm_state_versioned_exec(), with the new argument, and make the old function a wrapper for it with the new argument set to NULL.

In this case, I'm thinking it would be better to add versioned_params as a special key/value pair in params (using the entire versioned_params as the value). It's a little hacky, but it would simplify the code. Then, lrmd_api_exec() would just need to pull the value out before using params.